dpu_utils/utils/richpath.py

```diff
@@ -157,6 +162,25 @@ def exists(self) -> bool:
     def to_local_path(self) -> 'LocalPath':
         pass
 
+    def copy_from(self, source_path: 'RichPath') -> None:
+        if source_path.is_dir():
+            assert self.is_dir() or not self.exists(), 'Source path is a directory, but the target is a file.'
+            for file in source_path.iterate_filtered_files_in_dir('*'):
+                target_file_path = self.join(file.path[len(source_path.path):])

```

Comment:
I think this may be slightly wrong because `RichPath.create("foo").iterate_filtered_files_in_dir('*')` could produce things like `foo/bar`, and then the `[len("foo".path):]` would just be `/bar`. There's `RichPath.basename` for this case.

Reply:
I don't understand the comment (or at least I suspect that you are missing the recursive case). In any case, there was a but and I updated the code. Have a look now

Reply:
No, I think the bug you fixed was exactly the issue that I meant to raise. I'm not sure if `relpath` makes sense -- why is that better than just using the existing `basename`?

Reply:
Because basename will get the filename and this will break on recursive copies (`iterate_filtered_files_in_dir` is recursive, I recall).

So, if I understand your example, you have a `LocalPath` which points to the relative directory `foo`,`iterate_filtered_files_in_dir` will get all the files within that folder. We will then use their relative path in `foo` and copy them to same relative path as in `self`. I still don't see the corner case you mention :|